### ISSUE TITLE
Why do you need maven-reactor-plugin?

### ISSUE THREAD

**Issue by khmarbaise** [metasfresh/metasfresh#1275]
### Is this a bug or feature request?
In my opinion it is a bug if this plugin is really being used..cause it's long retired.

**Comment by metas-ts** [comment]
Hi @khmarbaise , 
I'm a jasper dilletante, so I can't really assess whether your statement is valid..
Could you give me some pointers to read up and form an opinion about maven-reactor-plugin?

So, if i'm taking your question at face value:

> Why do you need maven-reactor-plugin?

=> because we have a lot of modules, don't want to call each one indivdually and don't know how else to do it :-).
What whould be the alternative to maven-reactor?

Best regard
Tobi

PS: actually, today we talked about moving away not from the reactor, but from maven alltogether..but that's a bigger step of course..

**Comment by teosarca** [comment]
> PS: actually, today we talked about moving away not from the reactor, but from maven alltogether..but that's a bigger step of course..

+1000 :)

**Comment by khmarbaise** [comment]
Since Maven 3 (2010!) there is no need for the [maven-reactor-plugin](http://maven.apache.org/plugins/maven-reactor-plugin/) anymore. If you have a lot of modules means you can build them from the root via `mvn clean install` afterwards you can use: `mvn -pl moduleYouLikeToBuild install` if you are sure having no deps to other modules. If you like to let maven do the work you can change the call like this: `mvn -pl moduleYouLikeToBuild --also-make-dependents` which will also build the projects which are depending on the given module `moduleYouLikeToBuild` and other one. If you have modules which are required by the given module you can do this by using `mvn -pl moduleYouLikeToBuild --also-make`. 
But to use this you need to have a different project structure which you currently don't have...means a [hierarchical structure](https://github.com/khmarbaise/javaee) cause Indicators I've seen for this in your parent you have lines in there like this:

```xml
	<modules>
		<module>../de.metas.reactor</module>
		<module>../de.metas.reactor.ait</module>
		<module>../de.metas.esb</module>
	</modules>
```
This means the given pom should be in root level...The would make building and releases simpler etc.
Furthermore the module `...reactor` shows the same:
```xml
<modules>
		<module>../de.metas.assemblies</module>

		<module>../de.metas.migration</module>
		<module>../de.metas.scripts.rollout</module>
		
		<module>../de.metas.util</module>
		<module>../de.metas.monitoring</module>

		<module>../de.metas.adempiere.adempiere</module>
		...
```
This shows that the structure of your project is simply not very good.....

Apart from that the modules names look like OSGi naming...but I don't found any MANIFEST.MF entries which prove that...

Furthermore I don't see any `-SNAPSHOT` entry in your pom files but I see simply `1.0.0`...Does that mean you don't make SNAPSHOT's ?

Some other things I've seen is that you are duplicating the information in groupId in your artifactId like this:
```xml
<groupId>de.metas.adempiere.libero</groupId>
<artifactId>de.metas.adempiere.libero.liberoHR</artifactId>
```
I would go like this:
```xml
<groupId>de.metas.adempiere.libero</groupId>
<artifactId>libero.liberoHR</artifactId>
```
Because this would mean to have shorter module names which also means having shorter directory names for the appropriate modules etc.

Hint about Maven Command Line: http://maven.apache.org/ref/3.3.9/maven-embedder/cli.html

Kind regards
Karl Heinz Marbaise

**Comment by metas-ts** [comment]
Hi @khmarbaise ,
thanks a lot for your input. I directly bookmarked the link to the CLI options reference.

About the directory structure: it might be important as a background info to know that we rarely build from cmdline. We use eclipse with the m2e plugins to develop and jenkins to actually build...

> would make building and releases simpler etc.

..so I'm not sure what *concretely* would be simpler if we moved the directories around to create a less flat and more hierarchical structure. I can see some amount of effort for a single person to change it, then some more effort for the whole group to sync their workspaces, then probably some more to fix things that "somehow" worked because and now don't work anymore, but not a lot of value beeing added.
However, if we must change because there are some reall problems, it's of course very important to know that we should fall into the "hierarchical" direction and stop using the reactor.

> Apart from that the modules names look like OSGi naming...but I don't found any MANIFEST.MF entries which prove that...

We don't do a lot of OSGI, the names just reflect our preferences of the time those poms were created.
Basically it's the same thing here that I mentioned with the folder structure.

> Furthermore I don't see any -SNAPSHOT entry in your pom files but I see simply 1.0.0...Does that mean you don't make SNAPSHOT's ?

That's right, we don't make snapshots. We also don't deploy to our repository from local machines, only from jenkins. And jenkins in turn deploys "release" builds with discrete build numbers. We used to work with snaposhots for a long time of our history, but there was always some ammount of FUD  involved as to which particular version was taken at a certain time. Also the enforcer plugin's `banDuplicateClasses` didn't work well with snapshots.
Also, we don't really need the release process when we don't have snapshots. We can just pick the build we want and release it, and it already has "release" version numbers in it (actually we pretty much followed [this idea](https://axelfontaine.com/blog/maven-releases-steroids.html)).

Again, thx for your input, I appricate your thoughts. 
I would be very happy to also learn what you think about our build process (e.g. using maven version plugin to set version numbers etc.).

Best regards
Tobias
